src/sentry/ingest/transaction_clusterer/rule_validator.py

```diff
@@ -9,7 +10,46 @@ def __init__(self, rule: ReplacementRule, *, char_domain: Optional[str] = None)
         self._char_domain: Set[str] = set(char_domain) if char_domain else set("*/")
 
     def is_valid(self) -> bool:
-        return not self._is_all_stars()
+        if self._is_all_stars() or self._is_schema_and_all_stars():
+            return False
+        return True
 
     def _is_all_stars(self) -> bool:
-        return set(self._rule) <= self._char_domain
+        return self._is_string_all_stars(self._rule)
+
+    def _is_string_all_stars(self, string) -> bool:
+        return set(string) <= self._char_domain
+
+    def _is_schema_and_all_stars(self) -> bool:
+        """
+        Return true if the rule looks like a URL scheme and stars.
+
+        ## Examples
+        `http://*/*/**` -> `True`
+        `http://domain.com/*/**` -> `False`
+
+        Rules composed by the scheme and all stars provide the same value as all
+        stars: none. These rules are low quality and provide little renaming
+        value.
+
+        ## Assumptions
+        - Rules are URL-parsable, otherwise `False` is returned.
+        - URLs don't have queries or fragments, and thus they are ignored.
+        """
+        # urlparse doesn't validate the input and raises an exception if the
+        # string isn't a valid URL
+        try:
+            url = urlparse(self._rule)
```

Comment:
Wouldn't this check be easier with a Regex?

Reply:
That wouldn't require regex either, you use something like `startswith()` function. And we could eliminate the dependency here and also don't have to parse  (and maybe in some cases handle the exception).
What do you think?

Reply:
Actually a simple `split('://')` should be enough and then use the existing check for is all stars?

Reply:
- Although we're using `http` and `https` in the tests and examples, we don't want. to restrict ourselves to these two. `startswith` would require adding a check for each scheme we'd want to support.
- The regex approach assumes you'd have valid URLs in the input, which is something we can't rely on. We've seen `//` in combination with `/` in rules and transaction names, so a subdirectory in the path ending with a `:` would result in a false positive, see example below. Given the number of edge cases we've run into and `:` sometimes being used to identify variables (as a prefix, but I don't want to rely on too many assumptions), I wouldn't be surprised to see this come up as an issue in the future, and that's something I want to avoid.

```py
>>> '*/a://*'.split('://', 1)
['*/a', '*']
>>> urlparse('*/a://*')
ParseResult(scheme='', netloc='', path='*/a://*', params='', query='', fragment='')
```

What do you think? Happy to extend/update the test cases if I can provide more clarity too.

Reply:
We would have to check if the first item returned from the split matches valid characters of a protocol, only ascii? I feel like this is easier than all the extra netloc etc. checks, but maybe it isn't.

Reply:
I think `urlparse` and its explicit checks are simpler than doing it manually, plus the edge cases we may not think about at this point. Unless somebody feels strongly opinionated, I'd move forward with this approach. I'm happy to revisit this in the future if it turns out to not be the right choice.

Reply:
> We would have to check if the first item returned from the split matches valid characters of a protocol, only ascii?

I think `urlparse` does pretty much that (see linked code), so I think it's fine to use it instead of replicating that logic:

https://github.com/python/cpython/blob/57b7e52790ae56309832497a1ce17e3e731b920e/Lib/urllib/parse.py#L486-L490